airbyte-integrations/connectors/source-recharge/source_recharge/streams.py

```diff
@@ -0,0 +1,137 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from abc import ABC
+from enum import Enum
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional
+
+import pendulum
+import requests
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.requests_native_auth import TokenAuthenticator
+from airbyte_cdk.sources.utils.transform import TransformConfig, TypeTransformer
+
+
+class ApiVersion(Enum):
+    DEPRECATED = "2021-01"
+    MODERN = "2021-11"
+
+
+class RechargeTokenAuthenticator(TokenAuthenticator):
+    def get_auth_header(self) -> Mapping[str, Any]:
+        return {"X-Recharge-Access-Token": self._token}
+
+
+class Orders(HttpStream, ABC):
```

Comment:
I can see the transformer works fine for the migrated version, casting the `float` > `string` effectively, check the [expected_records](https://github.com/airbytehq/airbyte/blob/eae18a8bbb43776f624eceb11b8019878609734d/airbyte-integrations/connectors/source-recharge/integration_tests/expected_records_orders_modern_api.jsonl#L14), since they are aligned with the migrated version, so no matter what the API returns for that field, it should be cast to the string.